spec/controllers/sessions_controller_spec.rb

```diff
@@ -0,0 +1,71 @@
+require "rails_helper"
+
+describe SessionsController do
+  context "GET #new" do
+    context "without a password" do
+      it "does not set current user" do
+        user = create(:user)
+        session_params = { email: user.email }
+
+        post :create, params: { session: session_params }
+
+        expect(controller.current_user).to be nil
+      end
+
+      it "renders new" do
+        user = create(:user)
+        session_params = { email: user.email }
+
+        post :create, params: { session: session_params }
+
+        expect(response).to render_template(:new)
+      end
+
+      it "sets the flash[:alert]" do
+        user = create(:user)
+        session_params = { email: user.email }
+
+        post :create, params: { session: session_params }
+
+        expect(flash[:alert]).
+          to match I18n.
+          t("flash.actions.create.alert", resource_name: "Session")
+      end
+    end
+
+    context "with a password" do
+      it "sets current user" do
+        user = create(:user)
+        session_params = { email: user.email, password: user.password }
+
+        post :create, params: { session: session_params }
+
+        expect(controller.current_user.id).to eq user.id
```

Comment:
What do you think about changing this to be:
```
expect(controller.current_user).to eq user
```

Reply:
Done in sha: cc903dc